### DISCUSSION TITLE
Rx Crystal Oscillator Frequency Error measurement

### DISCUSSION THREAD

**Discussion by RMFPV** [ExpressLRS/ExpressLRS#1533] [Q&A] [answered]
Hi!
Is there the way for user to measure Crystal Oscillator (XO) Frequency Error?

**Comment by SunjunKim** [comment, marked as answer]
We have https://github.com/ExpressLRS/ExpressLRS/tree/SunjunKim-tx-freq-correction branch for testing this issue.

When a TX is flashed with this branch, it will enable a frequency correction on ch16 (=AUX12). Please assign a slider (or knob, wheel, whatever can do a continuous change) to ch16 

While an RX (should be on ver 2.x) is bound, by adjusting the CH16 value toward both positive and negative, you'll lose/recover the connection at some point. Please check which values it losses and recovers the connection (change the value slowly!).

When the values where you lose the connection is roughly symmetric (like, -60% and +70%), your units are OK.

If you lose the connection at around < +-15% at any of one side, your TX/RX pairs are having a risky frequency offset between them. 

When an RX is flashed with this branch, it applies a fixed -50 ppm frequency error correction. This will enable binding between deviated RXes (JHEMCU, Jumper&Matek w. caps on) to a common TXes, but as it's risky to fly with it it won't output control commands toward an FC.

**Comment by RMFPV** [comment]
Thank you Sunjun Kim!
This is very helpful! I will try that.

**Comment by RMFPV** [comment]
I did the test with Matek 2.4 D:
1440 - connection lost
up to max (2012) - connection available

I guess it means I need to remove caps. 

UPD
Without caps it works now down to 1270!
